store/sql_post_store.go

```diff
@@ -1355,3 +1355,35 @@ func (s SqlPostStore) GetPostsBatchForIndexing(startTime int64, limit int) Store
 
 	return storeChannel
 }
+
+func (s SqlPostStore) GetLastPostForChannelIfJoinLeave(channelId string) StoreChannel {
```

Comment:
Seems like this function might be doing a bit too much for a DBAL. Maybe it'd be better if we just made it `GetLastPostForChannel` and moved the rest of this logic into `app`. It's at least be more re-usable.